airflow/api_connexion/endpoints/xcom_endpoint.py

```diff
@@ -71,7 +71,7 @@ def get_xcom_entries(
         query = query.where(XCom.map_index == map_index)
     if xcom_key is not None:
         query = query.where(XCom.key == xcom_key)
-    query = query.order_by(DR.execution_date, XCom.task_id, XCom.dag_id, XCom.key)
+    query = query.order_by(DR.execution_date, XCom.task_id, XCom.map_index, XCom.dag_id, XCom.key)
```

Comment:
By the way I’m not quite sure why we order this way. Normally I’d order by dag_id before task_id?

Reply:
I'd even say:  dag_id, task_id, execution_date, map_index, key is the most reasonable ordering. I do not think it's likely someone bases their decisions on this ordering.

I'd be for changing it now, if we are adding the ordering now.

Reply:
Reading the implementation, I think the problem is if we put execution date in the middle it’d mess up the db index and affect performance. But we can order by run_id instead. Since we don’t technically promise ordering to the user in any way, I think I’ll just change this ordering to match the index.

Reply:
Perfect!